web/src/test/java/org/springframework/security/web/firewall/StrictHttpFirewallTests.java

```diff
@@ -723,6 +739,18 @@ public void getFirewalledRequestGetHeaderWhenNotAllowedHeaderNameThenException()
 		assertThatExceptionOfType(RequestRejectedException.class).isThrownBy(() -> request.getHeader("bad name"));
 	}
 
+	@Test
+	public void getFirewalledRequestGetHeaderWhenSetNotAllowedHeaderNamesThenException() {
+		this.firewall.setAllowedHeaderNames((name) -> !name.equals("bad name"));
+		this.firewall
+			.setAllowedHeaderNames(this.firewall.ALLOWED_HEADER_NAMES.and((name) -> !name.equals("worst name")));
```

Comment:
After the above changes, this will change to something closer to:

```java
this.firewall.setAllowedHeaderNames(
    StrictHttpFirewall.ALLOWED_HEADER_NAMES.and((name) -> !name.equals("worst name"));
```

Reply:
Done.